cluster-autoscaler/processors/provreq/processor_test.go

```diff
@@ -172,6 +175,55 @@ func TestRefresh(t *testing.T) {
 	}
 }
 
+func TestDeleteOldProvReqs(t *testing.T) {
+	now := time.Now()
+	tenDaysAgo := now.Add(-1 * 10 * 24 * time.Hour)
+	pr := provreqclient.ProvisioningRequestWrapperForTesting("namespace", "name-1")
+	additionalPr := provreqclient.ProvisioningRequestWrapperForTesting("namespace", "additional")
+
+	oldFailedPr := provreqclient.ProvisioningRequestWrapperForTesting("namespace", "failed")
+	oldExpiredPr := provreqclient.ProvisioningRequestWrapperForTesting("namespace", "expired")
+	oldFailedPr.CreationTimestamp = metav1.NewTime(tenDaysAgo)
+	oldExpiredPr.CreationTimestamp = metav1.NewTime(tenDaysAgo)
+	oldFailedPr.Status.Conditions = []metav1.Condition{
+		{
+			Type:               v1.Failed,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.NewTime(tenDaysAgo),
+			Reason:             "Failed",
+			Message:            "Failed",
+		},
+	}
+	oldFailedPr.Spec.ProvisioningClassName = v1.ProvisioningClassCheckCapacity
+	oldExpiredPr.Status.Conditions = []metav1.Condition{
+		{
+			Type:               v1.Provisioned,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.NewTime(tenDaysAgo),
+			Reason:             "Provisioned",
+			Message:            "",
+		},
+		{
+			Type:               v1.BookingExpired,
+			Status:             metav1.ConditionTrue,
+			LastTransitionTime: metav1.NewTime(tenDaysAgo),
+			Reason:             "Capacity is expired",
+			Message:            "",
+		},
+	}
+	oldExpiredPr.Spec.ProvisioningClassName = v1.ProvisioningClassCheckCapacity
+
+	client := provreqclient.NewFakeProvisioningRequestClient(nil, t, pr, additionalPr, oldFailedPr, oldExpiredPr)
+
+	processor := provReqProcessor{func() time.Time { return now }, 1, client, nil}
+	processor.refresh([]*provreqwrapper.ProvisioningRequest{pr, additionalPr, oldFailedPr, oldExpiredPr})
+
+	_, err := client.ProvisioningRequestNoCache(oldFailedPr.Namespace, oldFailedPr.Name)
```

Comment:
nit: I'd also assert that the other 2 behave differently (or that these 2 behave differently before the refresh).

Reply:
Done